### PR TITLE
User account forms improvements

### PR DESCRIPTION
**Pull request by kamalq97**
Changelog: 
- Added **forms.py** file that defines two forms that inherit from Django's built in authentication forms
- Updated `register` view in **views.py** to use a registration form based on Django's robust `UserCreationFrom`, which automatically checks if the data typed into the fields is valid (_This change fixes the bug where a violation of the database's unique constraint occurs when a user tries to register with an already existing email address_)
- Slightly updated `password_change_view` in **views.py** with a custom from based on Django's `PasswordChangeFrom`
- Updated **login.html**, **register.html**, and **change_password.html** to utilize new forms and with nicer looking CSS.

### CODE REVIEW

File: myapp/forms.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.contrib.auth.forms import UserCreationForm, PasswordChangeForm, AuthenticationForm
+
+
+class CustomRegistrationFrom(UserCreationForm):
+    username = forms.CharField(label='username', required=False, widget=forms.TextInput(attrs={'class': 'form-control', "placeholder": "Username", "rows": 1, "cols": 22}))
```
Comment (murradkh) [inline myapp/forms.py]: it's more comfortable to the eye to read short lines, line by line, the benefits:
1. convenience
2. readability
3. speed
4. consistency with the rest of the code in the project 

try it yourself, and see what's more comfortable :)

actually, maybe we need to create a new ticket for "Code Style" to add to `gitattributes` file, as you did before, in order to standardize, what do you think?

File: myapp/forms.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    email = forms.EmailField(label='email', max_length=100, widget=forms.TextInput(attrs={'class': 'form-control', "placeholder": "Email address", "rows": 1, "cols": 22}))
+    password1 = forms.CharField(label='password1', min_length=4, max_length=30, widget=forms.PasswordInput(attrs={'class': 'form-control', "placeholder": "Create password", "rows": 1, "cols": 20}))
+    password2 = forms.CharField(label='password2', min_length=4, max_length=30, widget=forms.PasswordInput(attrs={'class': 'form-control', "placeholder": "Confirm password", "rows": 1, "cols": 20}))
+
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'password1', 'password2']
```
Comment (murradkh) [inline myapp/forms.py]: the fields which already exist in the model `User`  no need to redefine (unless you have custom field attributes), for example:
```
class UserForm(forms.ModelForm):
    class Meta:
        model = User
        fields = ('username', 'email', 'password')
        widgets = {'username': forms.TextInput(attrs={
            "placeholder": "Username"
        }),
            "email": forms.EmailInput(attrs={
                "placeholder": "Email"
            }),
            "password": forms.PasswordInput(attrs={'placeholder': 'Password'})

        }
```

**Review by murradkh — COMMENTED** [review]
just small comments

File: myapp/forms.py (hunk 2 above)
Comment (kamalq97) [reply myapp/forms.py]: I am redefining them just to customize the widget CSS (purely for aesthetic reasons)

File: myapp/forms.py (hunk 1 above)
Comment (kamalq97) [reply myapp/forms.py]: Okay, I will fix! 👍

File: myapp/tests/test_forms.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+from django.contrib.auth.models import User
+from django.test import TestCase
+from myapp.forms import CustomRegistrationFrom, CustomChangePasswordForm
+
+
+class RegistrationFormTestCase(TestCase):
```
Comment (murradkh) [inline myapp/tests/test_forms.py:6]: Nice!
you covered many extreme-cases

File: myapp/tests/test_forms.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    'new_password1': 'randomexample',
+                    'new_password2': 'randomexample'}
+        form = CustomChangePasswordForm(self.test_user, data=test_data)
+        self.assertFalse(form.is_valid())
+        self.assertIn('old_password', form.errors.keys())
+
+    def test_empty_new_password(self):
+        test_data = {'old_password': 'firstuser123',
+                    'new_password1': '',
+                    'new_password2': ''}
```
Comment (murradkh) [inline myapp/tests/test_forms.py:64]: it's easier to read now after you shortened columns length

**Review by murradkh — APPROVED** [review]
Good job Kamal! you did a lot of stuff.

**Pull request merged**